frontend/src/Pages/Customer/ViewShops.js

```diff
@@ -0,0 +1,169 @@
+import { Link } from "react-router-dom";
+import '../../App.css';
+import LocationInput from '../LocationInput';
+import ROUTES from '../../routes';
+
+class ViewShops extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      shops: [],
+      searchQuery: "",
+      queryRadius: "",
+      showModal: true
+    }
+  }
+
+  search = async () => {
+
+    //Empty query implies browse intent
+    if (this.state.searchQuery === "") {
+      this.updateBrowseResults();
+      return;
+    }
+
+    // Use a default radius if no radius specified
+    if (this.state.queryRadius === "") {
+      this.setState({ queryRadius: "3km" });
+    }
+
+    const config = {
+      method: 'get',
+      url: ROUTES.api.get.index.search,
+      headers: {},
+      params: {
+        query: this.state.searchQuery,
+        queryRadius: this.state.queryRadius,
+        latitude: this.props.latitude,
+        longitude: this.props.longitude
+      }
+    };
+
+    const shopDetailsList = (await axios(config)).data;
+
+    this.setState({
+      "shops": shopDetailsList
+    })
+  }
+
+  updateBrowseResults = async () => {
+
+    // Use a default radius if no radius specified
+    if (this.state.queryRadius === "") {
+      this.setState({ queryRadius: "3km" });
+    }
+
+    const config = {
+      method: 'get',
+      url: ROUTES.api.get.index.browse,
+      headers: {},
+      params: {
+        queryRadius: this.state.queryRadius,
+        latitude: this.props.latitude,
+        longitude: this.props.longitude
+      }
+    };
+    const shopDetailsList = (await axios(config)).data;
+
+    this.setState({
+      "shops": shopDetailsList
+    })
+  }
+
+  onHide = () => {
+    this.setState({
+      showModal: false
+    });
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.latitude !== prevProps.latitude || (this.props.longitude !== prevProps.longitude)) {
+      this.updateBrowseResults();
+    }
+  }
+
+  componentDidMount() {
+    if (this.props.latitude !== null || (this.props.longitude !== null)) {
+      this.updateBrowseResults();
+    }
+  }
+
+  render() {
+    if (this.props.latitude == null || this.props.longitude == null) {
+      return (
+        <LocationInput setLocation={this.props.setLocation} show={this.state.showModal} onHide={this.onHide} />
+      );
+    } else {
+      return (
+        <Container className="mt-1 p-3">
+          <Form onSubmit={(e) => { e.preventDefault(); this.search(); }}>
+            <Form.Row className="align-items-center">
+              <Col xs={7}>
+                <Form.Control
+                  placeholder="Search Nearby"
+                  autoComplete="off"
+                  onChange={e => {
+                    this.setState({ searchQuery: e.target.value }, () => {
+                      if (this.state.searchQuery === "") {
+                        this.updateBrowseResults();
+                      }
+                    })
```

Comment:
Better to create a function for multi lines lambdas

Reply:
Done. Thank you!!